sdk/communication/communication-sms/test/smsClientWithToken.spec.ts

```diff
@@ -1,51 +1,66 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { record, Recorder, env } from "@azure/test-utils-recorder";
-import { SendRequest, SmsClient } from "../src/smsClient";
+import { env, isPlaybackMode, record, Recorder } from "@azure/test-utils-recorder";
+import { SmsSendRequest, SmsClient } from "../src/smsClient";
 import { assert } from "chai";
 import { isNode } from "@azure/core-http";
 import * as dotenv from "dotenv";
+import * as sinon from "sinon";
 import { parseConnectionString } from "@azure/communication-common";
 import { createCredential, recorderConfiguration } from "./utils/recordedClient";
+import { Uuid } from "../src/utils/uuid";
 
 if (isNode) {
   dotenv.config();
 }
 
-describe("SmsClientWithToken [Playback/Live]", async () => {
-  let recorder: Recorder;
+const isBrowser = new Function("try {return this===window;}catch(e){ return false;}");
 
-  beforeEach(async function() {
-    recorder = record(this, recorderConfiguration);
-  });
+if (!isBrowser()) {
+  describe("SmsClientWithToken [Playback/Live]", async () => {
+    let recorder: Recorder;
 
-  afterEach(async function() {
-    if (!this.currentTest?.isPending()) {
-      await recorder.stop();
-    }
-  });
+    beforeEach(async function() {
+      recorder = record(this, recorderConfiguration);
+      if (isPlaybackMode()) {
+        sinon.stub(Uuid, "generateUuid").returns("sanitized");
+        sinon.stub(Date, "now").returns(0);
+      }
+    });
+
+    afterEach(async function() {
+      if (!this.currentTest?.isPending()) {
+        await recorder.stop();
+      }
+      if (isPlaybackMode()) {
+        sinon.restore();
+      }
+    });
+
+    it("successfully issues a token for a user", async function() {
```

Comment:
This description doesn't sound right...

Reply:
I'll adjust this but yeah, i'll have a separate PR addressing our test overlap and unifying it